libraries/standard/mqtt/utest/mqtt_utest.c

```diff
@@ -554,42 +541,33 @@ void test_MQTT_Init_Invalid_Params( void )
     MQTTContext_t context;
     TransportInterface_t transport;
     MQTTFixedBuffer_t networkBuffer;
-    MQTTApplicationCallbacks_t callbacks;
 
-    setupCallbacks( &callbacks );
     setupTransportInterface( &transport );
 
     /* Check that MQTTBadParameter is returned if any NULL parameters are passed. */
-    mqttStatus = MQTT_Init( NULL, &transport, &callbacks, &networkBuffer );
-    TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
-
-    mqttStatus = MQTT_Init( &context, NULL, &callbacks, &networkBuffer );
+    mqttStatus = MQTT_Init( NULL, &transport, getTime, eventCallback, &networkBuffer );
     TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
 
-    mqttStatus = MQTT_Init( &context, &transport, NULL, &networkBuffer );
+    mqttStatus = MQTT_Init( &context, NULL, getTime, eventCallback, &networkBuffer );
     TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
 
-    mqttStatus = MQTT_Init( &context, &transport, &callbacks, NULL );
+    mqttStatus = MQTT_Init( &context, &transport, getTime, eventCallback, NULL );
     TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
 
-    /* Test if NULL is passed for any callbacks. */
-    callbacks.getTime = NULL;
-    mqttStatus = MQTT_Init( &context, &transport, &callbacks, &networkBuffer );
+    /* Test if NULL is passed for any functors. */
```

Comment:
I think we are just using function pointers here. Functors are slightly different from function pointers https://en.wikipedia.org/wiki/Function_pointer#Functors

Reply:
Oops, me coming from C++ land, am used to calling it functors (where I mostly used `std::function` objects 😛)  Will remove use of functor